lang/lang.go

```diff
@@ -267,3 +267,9 @@ func (obj *Lang) Close() error {
 	obj.wg.Wait()
 	return err
 }
+
+func (obj *Lang) Lint() error {
+	_, err := LexParse(obj.Input)
```

Comment:
This is actually the part which needs the most work... You probably want all of these to run:

https://github.com/purpleidea/mgmt/blob/master/docs/language-guide.md#stages

With the exception of actually running the function engine. All of those are needed to ascertain if a compile is successful.